### PR TITLE
debounce update events and add more `detail` to our `ide-event` (`CustomEvent`)

### PR DESCRIPTION
**Pull request by siddhantk232**
- Change the initial value of editor content to "loading...". This
  reduces the visual flickering that happens on initial load.

- `dispatch-event` (ftd) and `ide_dispatch_event` (js): add optional data parameter and rename the previous `data` arg to `name`. The new `data` is used to provide change details (file that changed and it's contents) from the `cm-editor`.

### CODE REVIEW

File: components/editor/editor.mjs [hunk 1]
```diff
@@ -26,10 +27,11 @@ class CMEditor extends HTMLElement {
 
         function update(vu) {
             self.documents[self.currentDocument] = vu.state;
+            window.ide_dispatch_event("save-unsaved-file", {file_name: self.currentDocument, content: vu.state.doc.toString()});
```
Comment (amitu) [inline components/editor/editor.mjs:30]: You can fire the event only if uv.some attribute says the editor content has changed.

File: components/editor/vars.ftd [hunk 2]
```diff
@@ -167,12 +167,8 @@ url: /foo/blog/images/first-image.jpg
 file-name: FASTN.ftd
 language: ftd
 
-customElements.define('cm-editor', CMEditor);
+loading...
```
Comment (amitu) [inline components/editor/vars.ftd:170]: I prefer we add another Boolean in vars and show a proper UI. Create a loading component that hides (using css class, if will destroy the editor, which contains state we want to preserve).

**Review by amitu — CHANGES_REQUESTED** [review]

File: components/editor/editor.mjs (hunk 1 above)
Comment (siddhantk232) [reply components/editor/editor.mjs]: Thanks! done in https://github.com/FifthTry/ui/pull/89/commits/77eeb06aaec92ce211e0cbed96754fc7b0880ee8

File: components/editor/vars.ftd (hunk 2 above)
Comment (siddhantk232) [reply components/editor/vars.ftd:170]: I agree but I would suggest we do this in another PR

**Pull request merged**